maestro-client/src/main/java/maestro/drivers/AndroidDriver.kt

```diff
@@ -398,9 +398,23 @@ class AndroidDriver(
             blockingStubWithTimeout.viewHierarchy(viewHierarchyRequest {})
         } catch (throwable: StatusRuntimeException) {
             val status = Status.fromThrowable(throwable)
-            if (status.code == Status.Code.DEADLINE_EXCEEDED) {
-                LOGGER.error("Timeout while fetching view hierarchy")
-                throw MaestroException.DriverTimeout("Android driver unreachable")
+            when (status.code) {
+                Status.Code.DEADLINE_EXCEEDED -> {
+                    LOGGER.error("Timeout while fetching view hierarchy")
+                    closed = true
+                    throw MaestroException.DriverTimeout("Android driver unreachable")
+                }
+                Status.Code.UNAVAILABLE -> {
```

Comment:
unavailable means the server is busy processing some other request, so it's technically retryable. Is that not the case in our case?

Reply:
Yup, this is retryable in this case. The only scenario where it’s not retryable is when the status is unavailable and cause is an IO exception. That indicates the server is no longer reachable, so we set close = true in that case.